pkg/kubevirt/data_volume.go

```diff
@@ -0,0 +1,118 @@
+package kubevirt
+
+import (
+	"context"
+	"fmt"
+
+	clouderrors "github.com/gardener/gardener-extension-provider-kubevirt/pkg/errors"
+
+	v1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+	cdi "kubevirt.io/containerized-data-importer/pkg/apis/core/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// DataVolumeManager manages Kubevirt DataVolume operations.
+type DataVolumeManager interface {
+	// CreateVolume creates a new kubevirt Data Volume from the data volume specs and in the passed namespace.
+	CreateVolume(ctx context.Context, secret *v1.Secret, name, namespace string, dvSpec cdi.DataVolumeSpec) error
+	// GetVolume fetches the specified volume by the passed name and namespace and return DataVolumeNotFoundError error in case of
+	// not found object error.
+	GetVolume(ctx context.Context, secret *v1.Secret, name, namespace string) (*cdi.DataVolume, error)
+	// ListDataVolumes lists all the Data Volumes which exists in the passed namespace.
+	ListDataVolumes(ctx context.Context, secret *v1.Secret, namespace string) (*cdi.DataVolumeList, error)
+	// DeleteVolume delete the DataVolume based on the passed name and namespace.
+	DeleteVolume(ctx context.Context, secret *v1.Secret, name, namespace string) error
+}
+
+type defaultDataVolumeManager struct {
+	client ClientFunc
+}
+
+// NewDefaultDataVolumeManager creates a new default manager with a k8s controller-runtime based on the sent kubecomfig.
+func NewDefaultDataVolumeManager(client ClientFunc) (DataVolumeManager, error) {
```

Comment:
This constructor is not used anywhere at the moment. Could you please add the code that would use it, or else clarify your intention for adding it?

Reply:
As discussed earlier this will be used once we decide how we will create those Data Volumes.

Reply:
Yes, thanks for clarifying.